cpp/dolfinx/mesh/utils.h

```diff
@@ -69,24 +69,48 @@ Eigen::Array<double, Eigen::Dynamic, 3, Eigen::RowMajor> midpoints(
     const mesh::Mesh& mesh, int dim,
     const Eigen::Ref<const Eigen::Array<int, Eigen::Dynamic, 1>>& entities);
 
-/// Compute indicies (local to the process) of all mesh entities that
-/// evaluate to true for the provided marking function. An entity is
-/// considered marked if the marker function evaluates true for all of
-/// the entities vertices.
-/// @cond Work around doxygen bug for std::function
+/// Compute indicies of all mesh entities that evaluate to true for the
+/// provided geometric marking function. An entity is considered marked
+/// if the marker function evaluates true for all of its vertices.
+///
 /// @param[in] mesh The mesh
 /// @param[in] dim The topological dimension of the entities to be
-///                considered
+///   considered
+/// @param[in] marker The marking function
+/// @returns List of marked entity indices, including any ghost indices
+///   (indices local to the process)
+Eigen::Array<std::int32_t, Eigen::Dynamic, 1> locate_entities(
+    const mesh::Mesh& mesh, const int dim,
+    const std::function<Eigen::Array<bool, Eigen::Dynamic, 1>(
+        const Eigen::Ref<const Eigen::Array<double, 3, Eigen::Dynamic,
+                                            Eigen::RowMajor>>&)>& marker);
+
+/// Compute indicies of all mesh entities that are attached to an owned
+/// boundary facet and evaluate to true for the provided geometric
+/// marking function. An entity is considered marked if the marker
+/// function evaluates true for all of its vertices.
+///
+/// @note For vertices and edges, in parallel this function will not
```

Comment:
Should this note be also in a python docstring? Its important.

Reply:
Added.